akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowDelaySpec.scala

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream.Attributes._
+import akka.stream.testkit.Utils._
+import akka.stream.testkit.scaladsl.TestSink
+import akka.stream.testkit.{ AkkaSpec, TestPublisher, TestSubscriber }
+import akka.stream.{ DelayOverflowStrategy, ActorMaterializer }
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+class FlowDelaySpec extends AkkaSpec {
+
+  implicit val materializer = ActorMaterializer()
+
+  "A Delay" must {
```

Comment:
add also a test of `Source(1 to 10).initialDelay()` which will delay the initial element of the source (but not others). Then you can see if your `delay` combinator properly adds time to that initial delay.